### DISCUSSION TITLE
How imToken wallet authorizes the acquisition of trx address ？

### DISCUSSION THREAD

**Discussion by elvis126** [consenlabs/webview#62] [Q&A] [answered]
I have switched imToken to TRX wallet, go to the "Browse" page to visit

const provider = new ethers.providers.Web3Provider(web3.currentProvider);
const signer = provider.getSigner();
const address = await signer.getAddress();

But the obtained address is still eth, not trx.

Excuse me, How to authorize to obtain trx address

**Comment by xwartz** [comment, marked as answer]
You should use [tronweb](https://github.com/tronprotocol/tronweb), not ethers.
